apps/pod-issuer-server/src/routes/deel.ts

```diff
@@ -0,0 +1,115 @@
+
+deel.post("/login", (req: Request, res: Response) => {
+  const inputs: { email: string; password: string } = req.body;
+  if (!inputs.email || !inputs.password) {
+    res.status(400).send("Missing query parameter");
+    return;
+  }
+
+  // In practice, get user information from database by email,
+  // Here for demo purposes, we'll allow any email with @zoo.com domain
+  if (!inputs.email.endsWith("@zoo.com")) {
+    res.status(401).send("Invalid email or password");
+    return;
+  }
+
+  // In practice, check if the encrypted password match
+  // This is just for demo purposes
+  if (!inputs.password.startsWith("zoo")) {
+    res.status(401).send("Invalid email or password");
+    return;
+  }
+
+  // Signing JWT, valid for 1 hour
+  const token = jwt.sign(
+    { email: inputs.email },
+    process.env.DEEL_EDDSA_PRIVATE_KEY!,
+    { algorithm: "HS512", expiresIn: "1h" }
+  );
+  res.send(token);
+});
+
+deel.post(
+  "/issue",
+  expressjwt({
+    secret: process.env.DEEL_EDDSA_PRIVATE_KEY!,
+    algorithms: ["HS512"]
+  }),
+  async (req: JWTRequest, res: Response) => {
+    const email = req.auth?.email;
+    if (!email) {
+      res.status(401).send("Unauthorized");
+      return;
+    }
+
+    const inputs: {
+      semaphoreCommitment: string;
+    } = req.body;
+
+    if (!inputs.semaphoreCommitment) {
+      res.status(400).send("Missing query parameter");
+      return;
+    }
+
+    try {
+      // We already issued paystub POD for this user, return the POD
+      const podStr = await getPaystubPODByEmail(email);
+      if (podStr !== null) {
+        const pod = POD.deserialize(podStr);
+        const owner = pod.content.asEntries().owner.value;
+        if (owner !== BigInt(inputs.semaphoreCommitment)) {
+          res
+            .status(400)
+            .send(
+              "Already issued POD for this user, but Semaphore Commitment doesn't match."
+            );
+          return;
+        }
+        res.status(200).json({ pod: podStr });
+        return;
+      }
+
+      const user = getDeelUserByEmail(email);
+      if (user === null) {
+        res.status(404).send("User not found");
+        return;
+      }
+
+      // For more info, see https://github.com/proofcarryingdata/zupass/blob/main/examples/pod-gpc-example/src/podExample.ts
+      const pod = POD.sign(
+        {
+          firstName: { type: "string", value: user.firstName },
+          lastName: { type: "string", value: user.lastName },
+          currentEmployer: { type: "string", value: "ZooPark" },
+          startDate: { type: "string", value: user.startDate },
+          annualSalary: { type: "int", value: BigInt(user.annualSalary) },
+          owner: {
+            type: "cryptographic",
+            value: BigInt(inputs.semaphoreCommitment)
+          }
+        } satisfies PODEntries,
+        process.env.DEEL_EDDSA_PRIVATE_KEY!
+      );
+      const serializedPOD = pod.serialize();
+
+      await savePaystubPOD(email, serializedPOD);
```

Comment:
Saving for future use is a reasonable use of caching and realistic behavior, though note that in this case if all the entries are identical, signing with the same key will generate an identical POD.

Reply:
yes, here I also don't want the user to change the semaphore commitment. otherwise, the nullifierHash check for each user can only get one loan won't matter.